azure-proto-core/Resources/UserAssignedIdentity.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace azure_proto_core.Resources
+{
+    using System;
+    using System.Text.Json;
+    using Azure.Core;
+
+    public class UserAssignedIdentity
+    {
+        public Guid? ClientId { get; set; }
+
+        public Guid? PrincipalId { get; set; }
+
+        public UserAssignedIdentity() { }
+
+        public UserAssignedIdentity(Guid clientId, Guid principalId)
+        {
+            ClientId = clientId;
+            PrincipalId = principalId;
+        }
+
+        public int CompareTo(UserAssignedIdentity other)
+        {
+            if (other == null)
+                return 1;
+
+            if (!ClientId.HasValue && !other.ClientId.HasValue)
+                return 0;
```

Comment:
What if ClientId is null and PrincipalId is not?  Is this a possible scenario? Or must PrincipalId have a value if ClientId does?

Reply:
the UserAssignedIdentity constructor doesn't allow for (Guid, null) because a Guid cannot be null. So we shouldn't ever have a case where ClientId has a value but PrincipalId does not